### PR TITLE
nuget packages are not ingesting sourceURL data automatically

### PR DESCRIPTION
**Pull request by johnbintz-tidelift**
## Current Production

Screenshot_20230405_091834

## Local Environment with Changes

Screenshot_20230405_091906

### CODE REVIEW

**Review by kjmahalingam — APPROVED** [review]
LGTM!

File: app/models/package_manager/nu_get.rb [hunk 1]
```diff
@@ -70,8 +76,32 @@ def self.project(name)
...
+      Zip::File.open_buffer(package_file(name, version)) do |zip|
+        zip.each do |file|
+          # the correct nuspec file will be at the root of the tree with the package name in the filename
+          file.get_input_stream { |io| nuspec = Ox.parse(io.read) } if file.name.downcase["#{name}.nuspec".downcase]
+        end
+      end
+
+      nuspec
+    rescue Zip::Error, Ox::ParseError
+      nil
```
[the remark this answers is not in the document]
Comment (johnbintz-tidelift) [reply app/models/package_manager/nu_get.rb:96]: `nil`, yes. Logging, sure I can add some.

File: app/models/package_manager/nu_get.rb [hunk 2]
```diff
@@ -70,8 +76,32 @@ def self.project(name)
...
+      StringIO.new(get_raw(url))
+    end
+
+    def self.nuspec(name, version)
+      nuspec = nil
+
+      Zip::File.open_buffer(package_file(name, version)) do |zip|
+        zip.each do |file|
+          # the correct nuspec file will be at the root of the tree with the package name in the filename
+          file.get_input_stream { |io| nuspec = Ox.parse(io.read) } if file.name.downcase["#{name}.nuspec".downcase]
```
Comment (jhan217) [inline app/models/package_manager/nu_get.rb:90]: Worth logging something if the expected file can't be found?

File: app/models/package_manager/nu_get.rb [hunk 3]
```diff
@@ -70,8 +76,35 @@ def self.project(name)
...
+        end
+      end
+
+      nuspec
+    rescue Zip::Error, Ox::ParseError => e
+      Rails.logger.error(
+        "Unable to download NuGet package for processing: NiGet/#{name}: #{e.message}"
```
Comment (djpowers) [inline app/models/package_manager/nu_get.rb:97]:
```suggestion
        "Unable to download NuGet package for processing: NuGet/#{name}: #{e.message}"
```
Possible typo?

**Comment by johnbintz-tidelift** [comment]
Yes, it's a typo. :)

@jhan217, package identifiers are case-insensitive: https://learn.microsoft.com/en-us/nuget/reference/nuspec#id

File: app/models/package_manager/nu_get.rb [hunk 4]
```diff
@@ -70,8 +76,40 @@ def self.project(name)
...
+
+      Zip::File.open_buffer(package_file(name, version)) do |zip|
+        zip.each do |file|
+          # the correct nuspec file will be at the root of the tree with the package name in the filename
+          # package identifiers are case-insensitive: https://learn.microsoft.com/en-us/nuget/reference/nuspec#id
+          file.get_input_stream { |io| nuspec = Ox.parse(io.read) } if file.name.downcase["#{name}.nuspec".downcase]
+        end
+      end
+
+      Rails.logger.error(
```
Comment (jhan217) [inline app/models/package_manager/nu_get.rb:95]: Should there be an if statement condition that needs to be met before logging this?

Comment (johnbintz-tidelift) [reply app/models/package_manager/nu_get.rb:95]: D'oh, yes.

**Review by kjmahalingam — APPROVED** [review]

**Review by kjmahalingam — APPROVED** [review]

**Review by macowie — APPROVED** [review]

**Pull request merged**